### PR TITLE
feat(api): new type definition and CE name

### PR DESCRIPTION
**Pull request by jason-capsule42**
BREAKING CHANGE: type is now a string, boolean type definitions removed

# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #17 
## Summary:

This change converts the component from `auro-alerts` to `auro-alert`. All references throughout the component updated accordingly.

Additionally, alert types are no longer separate boolean attributes but instead a single `type` string attribute.

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
-| Attribute | Type      | Description                   |
-|-----------|-----------|-------------------------------|
-| `fixed`   | `Boolean` | uses px values instead of rem |
+| Attribute     | Type      | Description                        |
+|---------------|-----------|------------------------------------|
+| `error`       | `Boolean` | Turns alert into error style       |
+| `fixed`       | `Boolean` | uses px values instead of rem      |
+| `information` | `Boolean` | Turns alert into information style |
+| `success`     | `Boolean` | Turns alert into success style     |
+| `warning`     | `Boolean` | Turns alert into warning style     |
```
Comment (blackfalcon) [inline docs/api.md:11]: The only attr that should be here is `fixed`. All other boolean attrs should be removed from the API docs.

File: docs/api.md [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
 | `hidden`         | `hidden`         | `Boolean` | If present, the component will be hidden both visually and from screen readers |
 | `hiddenAudible`  | `hiddenAudible`  | `Boolean` | If present, the component will be hidden from screen readers, but seen visually |
 | `hiddenVisually` | `hiddenVisually` | `Boolean` | If present, the component will be hidden visually, but still read by screen readers |
-| `information`    | `information`    | `Boolean` | Turns alert into information style               |
 | `noIcon`         | `noIcon`         | `Boolean` | Removes icon from alert UI                       |
 | `role`           | `role`           | `String`  | The role will be set based on type               |
-| `success`        | `success`        | `Boolean` | Turns alert into success style                   |
-| `warning`        | `warning`        | `Boolean` | Turns alert into warning style                   |
+| `type`           | `type`           | `String`  | Component will render visually based on which type value is set; currently supports `error`, `warning`, `success`, `information` |
```
Comment (blackfalcon) [inline docs/api.md:22]: How did this get here? I am not seeing an update to the src code to support this. 

I mean, it's correct, but when I run the API build locally, I get 

```
| `type`           | `type`           | `string`  |                                                  |
```

File: src/auro-alert.js [hunk 3]
```diff
@@ -74,37 +62,37 @@ class AuroAlerts extends AuroElement {
...
+    return this.noIcon
+      ? html``
+      : html`<div class="icon">${svg}</div>`
   }
 
   // function that renders the HTML and CSS into  the scope of the component
   render() {
     let output = html``;
-    const alertType = this.error || this.warning || this.information || this.success;
+    // const alertType = this.error || this.warning || this.information || this.success;
```
Comment (blackfalcon) [inline src/auro-alert.js:73]: delete

File: src/style.scss [hunk 4]
```diff
@@ -13,16 +13,16 @@
 @import "./node_modules/@alaskaairux/webcorestylesheets/dist/auroElement/auroElement";
 
 // container styles - defines the look of the outer custom component
-:host([error]),
-:host([warning]),
-:host([information]),
-:host([success]) {
+:host([type="error"]),
```
Comment (blackfalcon) [inline src/style.scss:16]: When you migrate these will need to be single quotes. Might as well get ahead of the game with this change.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: docs/api.md (hunk 2 above)
Comment (jason-capsule42) [reply docs/api.md:22]: I corrected the case of string to String and added the description manually.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Please update the JSdoc notes. 

I just did a test migration with the following steps;

1. legacy repo is on the `jbaker/redefineApi/#17` branch
2. created new `auro-alert` repo
3. migrated from legacy repo to new `auro-alert` repo
4. `master` was changed to `main`
5. all changes from `jbaker/redefineApi/#17` were consumed into the new `repoUpgrade` branch.

All the changes migrated without issue. 

```
[repoUpgrade 35442b1] refactor(generator migration): migrate repo to latest generator build
 44 files changed, 1651 insertions(+), 20542 deletions(-)
 create mode 100644 .editorconfig
 create mode 100644 .eslintrc
 delete mode 100644 .eslintrc.js
 delete mode 100644 .github/ISSUE_TEMPLATE/audit.md
 rename .github/ISSUE_TEMPLATE/{develop-design-story.md => dev-story.md} (76%)
 rewrite .github/ISSUE_TEMPLATE/general-support.md (67%)
 delete mode 100644 .github/ISSUE_TEMPLATE/new_wc.md
 delete mode 100644 .github/ISSUE_TEMPLATE/parent-design-story.md
 create mode 100644 .github/ISSUE_TEMPLATE/publish_component.md
 delete mode 100644 .github/ISSUE_TEMPLATE/review-design-story.md
 delete mode 100644 .github/ISSUE_TEMPLATE/symbol-design-story.md
 delete mode 100644 .github/ISSUE_TEMPLATE/user-story.md
 create mode 100644 .husky/.gitignore
 create mode 100755 .husky/commit-msg
 create mode 100755 .husky/pre-commit
 rewrite .stylelintrc (90%)
 create mode 100644 commitlint.config.js
 create mode 100644 demo/css/style.css
 create mode 100644 demo/demo.md
 create mode 100644 demo/style.scss
 delete mode 100644 docs/api.md
 delete mode 100644 docs/index.html
 delete mode 100644 karma.conf.js
 delete mode 100644 package-lock.json
 delete mode 100644 polymer.json
 create mode 100644 scripts/pre-commit.js
 create mode 100644 scripts/prepForBuild.js
 rewrite scripts/removeNonRemPlugin.js (70%)
 create mode 100644 web-test-runner.config.mjs
 ```
 
 Looking awesome!

File: docs/api.md [hunk 5]
```diff
@@ -1,24 +1,25 @@
...
+# auro-alert
 
 ## Attributes
 
-| Attribute | Type      | Description                   |
-|-----------|-----------|-------------------------------|
-| `fixed`   | `Boolean` | uses px values instead of rem |
+| Attribute     | Type      | Description                        |
+|---------------|-----------|------------------------------------|
+| `error`       | `Boolean` | Turns alert into error style       |
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply docs/api.md]: I realized why the build was messing up the api.md file and fixed it.

**Comment by jason-capsule42** [comment]
@blackfalcon This has been updated with the generator migration and should be ready for a final review and merging.

File: package.json [hunk 6]
```diff
@@ -7,82 +7,83 @@
...
   "repository": {
     "type": "git",
     "url": "https://github.com/AlaskaAirlines/auro-alert"
   },
   "main": "index.js",
   "license": "Apache-2.0",
   "dependencies": {
-    "chalk": "^4.1.0",
-    "lit-element": "^2.4.0"
+    "@alaskaairux/icons": "^4.7.0",
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply package.json:20]: Please see this issues https://github.com/AlaskaAirlines/WC-Generator/issues/159#issuecomment-868752049

File: scripts/prepForBuild.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+const indexFile = './build/index.html';
+
+// File destination.txt will be created or overwritten by default.
+let copyFiles = async function() {
+  fs.copyFile(`./dist/${bundle}`, `./build/${bundle}`, (err) => {
+    if (err) throw err;
+    console.log(`${bundle} was copied to ./build dir`);
+  });
+
+  fs.copyFile(`./demo/css/style.css`, `./build/css/style.css`, (err) => {
```
Comment (blackfalcon) [inline scripts/prepForBuild.js:12]: This is a bug in the generator https://github.com/AlaskaAirlines/WC-Generator/issues/160

Please update to point to the correct CSS file.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: scripts/prepForBuild.js (hunk 7 above)
Comment (jason-capsule42) [reply scripts/prepForBuild.js:12]: @blackfalcon shouldn't we be copying the .map file over as well?

Comment (blackfalcon) [reply scripts/prepForBuild.js]: no, the .map file will get generated as needed.

**Comment by blackfalcon** [comment]
With the migration, the new `main` branch needs to be pushed to the remote repo as well. The PR should be updated to point to `main` as well. 

This is needed because the definitions in the `settings.yml` are looking for `main` not `master`. That's why the tests haven't fired yet.

**Pull request merged**